chaosaws/ec2/actions.py

```diff
@@ -17,3 +17,54 @@ def stop_instance(instance_id: str, force: bool = False,
     """
     client = aws_client('ec2', configuration, secrets)
     return client.stop_instances(InstanceIds=[instance_id], Force=force)
+
+
+def stop_random_instance(force: bool=False,
+                         configuration: Configuration=None,
+                         secrets: Secrets=None) -> AWSResponse:
+    """
+    Stop a random EC2 instance.
+    """
+    client = aws_client('ec2', configuration, secrets)
+    res = client.describe_instances()
+    x = random.randrange(0, len(res['Reservations']))
+    instance_id = res['Reservations'][x]['Instances'][0]['InstanceId']
+    return client.stop_instances(InstanceIds=[instance_id], Force=force)
+
+
+def stop_random_instance_az(az: str, force: bool=False,
+                            configuration: Configuration=None,
+                            secrets: Secrets=None) -> AWSResponse:
+    """
+    Stop a random EC2 instance in a given availability zone.
+    """
+    client = aws_client('ec2', configuration, secrets)
+    filters = [{'Name': 'availability-zone', 'Values': [az]}]
+    res = client.describe_instances(Filters=filters)
```

Comment:
Should we a conditional here for when `res` is empty?

Maybe we should fail the activity? Something like:

```
if not res:
   raise FailedActivity("could not find az : {}".format(az)
```

Reply:
Hmm, not sure if res could be empty (really not sure, but maybe). However, instance_ids could be '[ ]' and stop_instance called on empty list lead to an error so I'll check this.

Reply:
Just checked, with a non-existing AZ, describe_instances(Filters=.......) return {"Reservations": []} so shouldn't be a problem since the for each will let instance_ids empty, agree?.